httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -313,11 +314,14 @@ public URI build() throws URISyntaxException {
         return new URI(buildString());
     }
 
-    private String buildString() {
+    private String buildString() throws URISyntaxException {
         final StringBuilder sb = new StringBuilder();
         if (this.scheme != null) {
             sb.append(this.scheme).append(':');
         }
+        if (URIScheme.HTTPS.same(scheme) && (host == null || host.isEmpty())) {
```

Comment:
@arturobernalg Could you please use `TextUtils#isBlank` here for the host check? Looks good otherwise.

Reply:
@ok2c you're right. Changed.